Mage.Server.Plugins/Mage.Player.Human/src/mage/player/human/HumanPlayer.java

```diff
@@ -1398,6 +1397,40 @@ && lookAtFaceDownCard((Card) object, game, useableAbilities.size())) {
         return false;
     }
 
+    private boolean skipButtonActivated() {
+        return passedAllTurns || passedUntilEndStepBeforeMyTurn || passedTurn || passedUntilEndOfTurn
+                || passedUntilNextMain;
+    }
+
+    private boolean quickStopForLands(Game game) {
+        if (skipButtonActivated()) {
+            Battlefield battlefield = game.getBattlefield();
+            for (Permanent land : battlefield.getAllActivePermanents(filterLand, playerId, game)) {
+                if (land.canUseActivatedAbilities(game)) {
+                    String name = land.getName();
+                    if (name.equals("Maze of Ith") || name.equals("Mystifying Maze") || name.equals("Spires of Orazca")) {
+                        // Find valid targets
+                        for (Permanent attacker : battlefield.getAllActivePermanents(filterAttack, game)) {
+                            if (attacker.canBeTargetedBy(land, playerId, game)) {
+                                return true;
+                            }
+                        }
+                    } else if (name.equals("Maze of Shadows")) {
+                        // Find valid targets
+                        for (Permanent attacker : battlefield.getAllActivePermanents(filterAttack, game)) {
+                            if (attacker.hasAbility(ShadowAbility.getInstance(), game)
+                                    && attacker.canBeTargetedBy(land, playerId, game)) {
+                                return true;
+                            }
+                        }
+                    }
+                    // TODO: Add untapped Manlands (via detection of BecomesCreatureSourceEffect usage?)
```

Comment:
Must add uncompleted tasks and use cases in the PR description

Reply:
Done. Added above too: This can also be applied to untapped Manlands but I don't think there's an easy way to get `BecomesCreatureSourceEffect` usage so I think this should be saved for another PR.

I'll look at the other comments later. Thanks!